### DISCUSSION TITLE
How to keep the URI unchanged

### DISCUSSION THREAD

**Discussion by txwdzxq** [coder/code-server#6106] [Q&A] [answered]
In previous versions
code-server: v3.4.1
VS Code: v1.45.1

Open workspce can keep the uri unchanged
such as
www.coder.com
![image](https://user-images.githubusercontent.com/36099482/227743228-56699eab-02ce-4949-867d-42e07c76cc07.png)

But in the current version
code-server: v4.11.0
Code: 1.76.1

The uri will redirect to
www.coder.com/? workspace=/home/coder/project.code-workspace
![image](https://user-images.githubusercontent.com/36099482/227743199-209e221f-35fe-4a5b-94a0-9fbfc56c849f.png)

I don't want the redirect
just url
How should I configure it?

**Comment by code-asher** [comment, marked as answer]
It is not currently possible to change this behavior.

**Reply by txwdzxq** [reply]
But the previous version supported doing this. What was the reason for changing it to the current state? I don't want to expose my directory structure in the address bar. Also, it seems that when I log in on another computer, the opened files are not synchronized. Is there any way to access my last working state every time I log in, regardless of the situation? I don't want to reopen the last edited file every time I log in on a different computer, especially when the directory is deep.

**Reply by code-asher** [reply]
If I recall correctly upstream in VS Code they removed the folder/workspace parameters you could pass in so the only way to specify those on our end (without doing more patching to Code) was to provide them through the URL.

The opened file behavior is another upstream behavior that changed, it gets saved in browser storage now so it is unique per browser.  I did look into trying to patch it so it would save to the disk again but it looked non-trivial.  Might be worth another look to see if it has become any easier.  Presumably it reproduces in Codespaces; if so we could try reporting it upstream to see if they will consider changing the behavior back.

**Reply by txwdzxq** [reply]
okey, thanks :)
